### ISSUE TITLE
An event that only triggers on blur and when valid

### ISSUE THREAD

**Issue by cpauwels** [wearebraid/vue-formulate#304]
It is currently not the best experience to use `vue-formulate` for inputs that are not tied to a form or save button. For instance, in a web app, changing the name of an item when the input is blurred.

Ideally, the `FormulateInput` has an event that triggers on blur, but only if the input value is valid. That way an API call could be made to update the backend with the value of the event. And since it would only trigger when the input has no errors, we wouldn't have to use `hasValidationErrors()` with the input `ref`, or keeping a `hasErrors` in the `data` that is updated on `@validation` events.

**Comment by justin-schroeder** [comment]
That is a very very specific event you're looking for there, a compound of `event + state`. Perhaps it might make more sense if we had a better way to know an input's state at any given time and you could check it in your blur event.

```vue
<FormulateInput
  @blur="ctx => ctx.isValid ? doThings() : dontDoThings()"
/>
```

That seems reasonable right? Then we don't need to come up with so many compound-events. I'll tag this for feature release, but feel free to comment. Suggestions welcome.

**Comment by cpauwels** [comment]
The solution you're suggesting would be ideal! 👍

**Comment by justin-schroeder** [comment]
@cpauwels this is now included in `release/2.5.0`. The event name is `@blur-context` and the payload is the full context object — so my example up above would apply with only a slight change to the event name:

```vue
<FormulateInput
  @blur-context="ctx => ctx.isValid ? doThings() : dontDoThings()"
/>
```
